### PR TITLE
feat: Enable remaining Spark 3.5.1 tests

### PR DESCRIPTION
**Pull request by andygrove**
## Which issue does this PR close?

Closes https://github.com/apache/datafusion-comet/issues/617

### CODE REVIEW

File: common/src/main/scala/org/apache/comet/CometConf.scala [hunk 1]
```diff
@@ -289,7 +289,7 @@ object CometConf extends ShimCometConf {
           "why a query stage cannot be executed natively. Set this to false to " +
           "reduce the amount of logging.")
       .booleanConf
-      .createWithDefault(true)
+      .createWithDefault(false)
```
Comment (andygrove) [inline common/src/main/scala/org/apache/comet/CometConf.scala:292]: Logging was too verbose so I changed this back to be off by default

File: dev/diffs/3.5.1.diff [hunk 2]
```diff
@@ -2120,29 +2120,17 @@ index 746f289c393..bc01ffd52ea 100644
...
+@@ -1029,15 +1055,23 @@ abstract class BucketedReadSuite extends QueryTest with SQLTestUtils with Adapti
            Seq(true, false).foreach { aqeEnabled =>
              withSQLConf(SQLConf.ADAPTIVE_EXECUTION_ENABLED.key -> aqeEnabled.toString) {
                val plan = sql(query).queryExecution.executedPlan
 -              val shuffles = collect(plan) { case s: ShuffleExchangeExec => s }
 +              val shuffles = collect(plan) {
-+            case s: ShuffleExchangeLike => s
-+          }
++                case s: ShuffleExchangeLike => s
++              }
```
Comment (kazuyukitanimura) [inline dev/diffs/3.5.1.diff:2130]: nit: this part of diff seems unnecessary

**Review by kazuyukitanimura — APPROVED** [review]

**Review by huaxingao — APPROVED** [review]

File: dev/diffs/3.5.1.diff (hunk 2 above)
Comment (andygrove) [reply dev/diffs/3.5.1.diff:2130]: this was just fixing incorrect formatting

Comment (andygrove) [reply dev/diffs/3.5.1.diff:2130]: nm, I see what you mean now

Comment (andygrove) [reply dev/diffs/3.5.1.diff]: fixed

**Pull request merged**